src/frontend/src/functions/auth.tsx

```diff
@@ -119,36 +113,47 @@ export function handleReset(navigate: any, values: { email: string }) {
 }
 
 /**
- * Check login state, and redirect the user as required
+ * Check login state, and redirect the user as required.
+ *
+ * If there is no token available in the session, then the user is *not* logged in.
+ * If there is a token available, we need to test it!
  */
 export function checkLoginState(
   navigate: any,
   redirect?: string,
   no_redirect?: boolean
 ) {
+  if (!useSessionState.getState().hasToken()) {
+    // No token available - redirect to the login page
+    if (!no_redirect) navigate('/login');
+    return;
+  }
+
+  // There *is* a token available: Test if it is valid
   api
-    .get(apiUrl(ApiEndpoints.user_token), {
-      timeout: 2000,
-      params: {
-        name: 'inventree-web-app'
-      }
+    .get(apiUrl(ApiEndpoints.user_me), {
+      timeout: 2500
```

Comment:
@matmair ok, that's a fair point. The large issue with "global login" is that when you press "logout", it does not actually logout, because there is still a session CRSF token available. I'll see if there's a way to fix this.

Reply:
I have now added in explicit support for the CSRF session cookie, allowing login based on that. It works quite smoothly now if you are already logged in via PUI!